x-pack/plugins/security_solution/public/common/components/autocomplete/helpers.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import dateMath from '@elastic/datemath';
+import { EuiComboBoxOptionOption } from '@elastic/eui';
+
+import { IFieldType } from '../../../../../../../src/plugins/data/common';
+import { Ipv4Address } from '../../../../../../../src/plugins/kibana_utils/public';
+import {
+  EXCEPTION_OPERATORS,
+  isOperator,
+  isNotOperator,
+  existsOperator,
+  doesNotExistOperator,
+} from './operators';
+import { GetGenericComboBoxPropsReturn, OperatorOption } from './types';
+
+export const getOperators = (field: IFieldType | undefined): OperatorOption[] => {
+  if (field == null) {
+    return [isOperator];
+  } else if (field.type === 'boolean') {
+    return [isOperator, isNotOperator, existsOperator, doesNotExistOperator];
+  } else if (field.type === 'nested') {
+    return [isOperator];
+  } else {
+    return EXCEPTION_OPERATORS;
+  }
+};
+
+export function validateParams(params: string, type: string) {
+  switch (type) {
+    case 'date':
+      const moment = dateMath.parse(params);
+      return Boolean(moment && moment.isValid());
+    case 'ip':
+      try {
+        return Boolean(new Ipv4Address(params));
```

Comment:
I noticed type `ip` can by v4 or v6, but didn't see a good check for v6. May just add a regex for it.